src/solvers/dg_unstructured_quad/dg_2d.jl

```diff
@@ -0,0 +1,341 @@
+        interfaces.u[1, v, i, interface] = u[v, i, 1, primary_element]
+      end
+    elseif primary_side == 2
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[1, v, i, interface] = u[v, nnodes(dg), i, primary_element]
+      end
+    elseif primary_side == 3
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[1, v, i, interface] = u[v, i, nnodes(dg), primary_element]
+      end
+    else # primary_side == 4
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[1, v, i, interface] = u[v, 1, i, primary_element]
+      end
+    end
+
+    if secondary_side == 1
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[2, v, i, interface] = u[v, i, 1, secondary_element]
+      end
+    elseif secondary_side == 2
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[2, v, i, interface] = u[v, nnodes(dg), i, secondary_element]
+      end
+    elseif secondary_side == 3
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[2, v, i, interface] = u[v, i, nnodes(dg), secondary_element]
+      end
+    else # secondary_side == 4
+      for i in eachnode(dg), v in eachvariable(equations)
+        interfaces.u[2, v, i, interface] = u[v, 1, i, secondary_element]
+      end
+    end
+  end
+
+  return nothing
+end
+
+
+# compute the numerical flux interface coupling between two elements on an unstructured quadrilateral mesh
+function calc_interface_flux_unstructured_quad!(surface_flux_values::AbstractArray{<:Any,4},
+                              nonconservative_terms::Val{false}, equations, dg::DG, cache)
+  @unpack surface_flux = dg
+  @unpack u, start_index, inc_index, element_ids, element_side_ids = cache.interfaces
+  @unpack normals, tangents, scaling = cache.elements
+
+
+  @threaded for interface in eachinterface(dg, cache)
+    # Get neighboring elements
+    primary_element   = element_ids[1, interface]
+    secondary_element = element_ids[2, interface]
+
+    # Get the local side id on which to compute the flux
+    primary_side   = element_side_ids[1, interface]
+    secondary_side = element_side_ids[2, interface]
+
+    # initial index for the coordinate system on the secondary element
+    j = start_index[interface]
+
+    # loop through the primary element coordinate system and compute the interface coupling
+    for i in eachnode(dg)
+      # pull the primary and secondary states from the boundary u values
+      u_ll = u[1, :, i, interface]
+      u_rr = u[2, :, j, interface]
+
+      # pull the directional vectors and scaling factors
+      #   Note! this assumes a conforming approximation, more must be done in terms of the normals
+      #         and tangents for hanging nodes and other non-conforming approximation spaces
+      n_vec   = normals[ :, i, primary_side, primary_element]
+      t_vec   = tangents[:, i, primary_side, primary_element]
```

Comment:
This allocates. Thus, if we keep this kind of storage layout, we probably need some accessors that return an `SVector` to get useful performance (maybe just leave a TODO for now?)